src/main/java/de/zalando/aruha/nakadi/service/CursorsCommitService.java

```diff
@@ -112,16 +116,41 @@ private void createSubscriptionInZkIfNeeded(final Subscription subscription) thr
 
                             subscriptionClient.fillEmptySubscription(subscriptionOffsets);
                         }
-                    } catch (Exception e) {
+                    } catch (final Exception e) {
                         atomicReference.set(e);
                     }
                 });
             }
-        } catch (Exception e) {
+        } catch (final Exception e) {
             atomicReference.set(e);
         }
         if (atomicReference.get() != null) {
-            throw new ServiceUnavailableException("Error communicating with zookeeper", atomicReference.get());
+            throw new ServiceUnavailableException(ERROR_COMMUNICATING_WITH_ZOOKEEPER, atomicReference.get());
+        }
+    }
+
+    public List<Cursor> getSubscriptionCursors(final String subscriptionId) throws NakadiException {
+        final Subscription subscription = subscriptionRepository.getSubscription(subscriptionId);
+        final String eventTypeName = subscription.getEventTypes().iterator().next();
+        final String topic = eventTypeRepository.findByName(eventTypeName).getTopic();
+        final String partitionsPath = MessageFormat.format(PATH_ZK_PARTITIONS, subscriptionId, topic);
+
+        try {
+            return zkHolder.get().getChildren().forPath(partitionsPath).stream()
+                    .map(partition -> createCursor(subscriptionId, topic, partition))
+                    .collect(Collectors.toList());
+        } catch (final Exception e) {
+            throw new ServiceUnavailableException(ERROR_COMMUNICATING_WITH_ZOOKEEPER, e);
+        }
+    }
+
+    private Cursor createCursor(final String subscriptionId, final String topic, final String partition) {
+        try {
+            final String offsetPath = MessageFormat.format(PATH_ZK_OFFSET, subscriptionId, topic, partition);
+            final String currentOffset = new String(zkHolder.get().getData().forPath(offsetPath), CHARSET_UTF8);
+            return new Cursor(partition, currentOffset);
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Because of that I have added 'throws' declaration. I think it looks cleaner that way.